src/PrestaShopBundle/EventListener/Admin/Context/ShopContextListener.php

```diff
@@ -183,4 +206,34 @@ private function redirectShopContext(RequestEvent $requestEvent): ?RedirectRespo
             ['setShopContext', 'conf']
         ));
     }
+
+    /**
+     * @throws ReflectionException
+     */
+    private function verifyRouteAttribute(Request $request): ?ShopConstraint
+    {
+        try {
+
+            $routeInfo = $this->router->match($request->getPathInfo());
+            if (empty($routeInfo['_controller'])) {
+                return null;
+            }
+
+            $controller = $routeInfo['_controller'];
+            [$className, $methodName] = explode('::', $controller);
```

Comment:
You should check that controller and methodName are not empty maybe?

Reply:
For me this cannot happen, except in the context of the API where the `_controller` does not contain '::' because API platform fills this field differently. And if that happened we would get a `ReflectionException`

Reply:
There also other notations for controller in symfony, but I guess you're right Worst case the exception is triggered and caught